File: pet-clinic-data/src/main/java/com/sachet/sfgpetclinic/services/PetTypeService.java
package com.sachet.sfgpetclinic.services;

import com.sachet.sfgpetclinic.model.PetType;

import java.util.Set;

public interface PetTypeService {

    Set<PetType> findAll();

    PetType findById(Long id);

    PetType save(PetType petType);

    void delete(PetType petType);

    void deleteById(Long id);

}

File: src/main/kotlin/com/sachet/sfgpetclinic/service/PetTypeService.kt
package com.sachet.sfgpetclinic.service

import com.sachet.sfgpetclinic.model.PetType

interface PetTypeService {

    fun findById(id: Long): PetType
    fun save(petType: PetType): PetType
    fun findAll(): List<PetType>
    fun delete(petType: PetType)
    fun deleteById(id: Long)

}